File: backspace.cpp
#include<string>
#include<iostream> 
#include<vector>
#include <algorithm>
using namespace std;
int main()
{
	string line;
	cin >> line;
	vector<char> chars;
	string output;
	
	for (int i = 0; i < line.size(); i++) {
		if( line[i] == '<'){
			chars.pop_back();
		}
		else {
			chars.push_back(line[i]);
		}
	}
	for (int i = 0; i < chars.size(); i++) {
		output += chars[i];
	}
	if (output.size()) {
		cout << output;
	}
	return 0;
}

File: backspace.py
l = input()
output = ""
line = [char for char in l]
i = 1
while i < len(line):
    if line[i] == '<':
        line.pop(i)
        line.pop(i-1)
        i -= 1;
    else:
        i +=1
for x in line:
    output += x
if(len(line) > 0):
    print(output)
